ideascube/serveradmin/tests/test_views.py

```diff
@@ -642,3 +642,29 @@ def test_staff_forgets_non_existing_connection(mocker, staffapp):
     assert u'No known Wi-Fi networks' not in res.unicode_body
     assert u'my home network' in res.unicode_body
     assert u'random open network' in res.unicode_body
+
+
+@pytest.mark.usefixtures('installed_packages_catalog')
+def test_staff_should_see_packages_with_cards(staffapp, systemuser):
+    from ideascube.configuration import set_config
+    set_config('ideascube', 'displayed-package-ids', ['test.package1'], systemuser)
+    res = staffapp.get(reverse("server:cards"), status=200)
+    assert 'Test package1' in res.unicode_body
+    assert 'Test package2' in res.unicode_body
+    assert 'Test package3' not in res.unicode_body
+
+    form = res.forms['cards']
+    inputs = form.fields['visible']
+    for _input in inputs:
+        should_be_checked = (_input._value == 'test.package1')
```

Comment:
>> Why don't you give an id to each input? That would allow you to identify them uniquely.
> Cause this id would be used only for testing.

Well, no, it could be used also in the CSS, the Javascript, ... That's really very standard HTML.

> And all the logic is base on the value attribute.

And maybe that's the problem? `value` doesn't have to be unique in an HTML page, it doesn't even have to be unique in a form. Using `value` to identify (see the first 2 letters in that word?) an object seems wrong to me.

It would be like trying to identify an element of a dictionary by its value, rather than by its key.

I really don't see why you wouldn't be testing the right thing here.

You could set `id="pkg_{{ pkg.id }}" value="{{ pkg.id }}"` in your HTML, then in the tests use `id` to identify which ones should be checked, and even verify that `id == 'pkg_%s' % value` if you want.

There's really no reason you can't bind the two like this.